internal/hmyapi/apiv1/blockchain.go

```diff
@@ -728,8 +778,52 @@ func (s *PublicBlockChainAPI) GetDelegationsByDelegator(ctx context.Context, add
 	return result, nil
 }
 
+// GetDelegationsByDelegatorByBlockNumber returns list of delegations for a delegator address at given block number
+func (s *PublicBlockChainAPI) GetDelegationsByDelegatorByBlockNumber(
+	ctx context.Context, address string, blockNum rpc.BlockNumber,
+) ([]*RPCDelegation, error) {
+	if err := s.isBeaconShard(); err != nil {
+		return nil, err
+	}
+	if err := s.isBlockGreaterThanLatest(blockNum); err != nil {
+		return nil, err
+	}
+	delegatorAddress := internal_common.ParseAddr(address)
+	block, err := s.b.BlockByNumber(ctx, rpc.BlockNumber(blockNum))
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not retrieve the block information for block number: %d", blockNum)
+	}
+	validators, delegations := s.b.GetDelegationsByDelegatorByBlockNumber(delegatorAddress, block)
+	result := []*RPCDelegation{}
+	for i := range delegations {
+		delegation := delegations[i]
+
+		undelegations := []RPCUndelegation{}
```

Comment:
since size is known, use `make` and assign to index instead of literal and append.